### PR TITLE
Add breaking api changes from ros2_control to hardware_interface

### PR DESCRIPTION
**Pull request by livanov93**
Needed after ros-controls/ros2_control#503.

### CODE REVIEW

File: ur_robot_driver/include/ur_robot_driver/hardware_interface.hpp [hunk 1]
```diff
@@ -76,24 +75,20 @@ class URPositionHardwareInterface : public hardware_interface::SystemInterface
...
 
-  status get_status() const final
-  {
-    return status_;
-  }
-
   std::string get_name() const final
   {
     return info_.name;
   }
```
Comment (destogl) [inline ur_robot_driver/include/ur_robot_driver/hardware_interface.hpp:87]: This should be now implemented in the [default implementation](https://github.com/ros-controls/ros2_control/blob/04fa19b50e9e67cf7e6d7cc5c7ba000e435440f9/hardware_interface/include/hardware_interface/system_interface.hpp#L183). It is safe to delete.

```suggestion
```

File: ur_robot_driver/include/ur_robot_driver/hardware_interface.hpp [hunk 2]
```diff
@@ -45,10 +44,10 @@
 
 // ROS
 #include "rclcpp/macros.hpp"
+#include "rclcpp_lifecycle/state.hpp"
 
 using hardware_interface::HardwareInfo;
 using hardware_interface::return_type;
```
Comment (destogl) [inline ur_robot_driver/include/ur_robot_driver/hardware_interface.hpp:50]: I propose to remove this using. There are only a few places where they are useful, and it is clearer when full type reference is written-out.

```suggestion
```

File: ur_robot_driver/include/ur_robot_driver/hardware_interface.hpp [hunk 3]
```diff
@@ -126,7 +121,6 @@ class URPositionHardwareInterface : public hardware_interface::SystemInterface
   void updateNonDoubleValues();
 
   HardwareInfo info_;
```
Comment (destogl) [inline ur_robot_driver/include/ur_robot_driver/hardware_interface.hpp:123]: Not needed anymore, [see](https://github.com/ros-controls/ros2_control/blob/04fa19b50e9e67cf7e6d7cc5c7ba000e435440f9/hardware_interface/include/hardware_interface/system_interface.hpp#L198)

```suggestion
```

**Review by destogl — APPROVED** [review]
Three small optimization proposals. Otherwise, all good! Thanks!

File: ur_robot_driver/include/ur_robot_driver/hardware_interface.hpp (hunk 3 above)
Comment (livanov93) [reply ur_robot_driver/include/ur_robot_driver/hardware_interface.hpp]: d4a1834

File: ur_robot_driver/include/ur_robot_driver/hardware_interface.hpp (hunk 2 above)
Comment (livanov93) [reply ur_robot_driver/include/ur_robot_driver/hardware_interface.hpp]: d4a1834

File: ur_robot_driver/include/ur_robot_driver/hardware_interface.hpp (hunk 1 above)
Comment (livanov93) [reply ur_robot_driver/include/ur_robot_driver/hardware_interface.hpp]: d4a1834

**Pull request merged**